components/tidb_query_datatype/src/codec/mysql/json/path_expr.rs

```diff
@@ -536,18 +515,17 @@ mod tests {
             ),
             (
                 "$.\"\\u33\"",
-                // TODO: pass the position in the unquote unicode error
-                Some("Invalid unicode, byte len too short"),
+                Some("Invalid JSON path expression. The error is around character position 3."),
                 None,
             ),
             (
                 "$**",
-                Some("Invalid JSON path expression. The error is around character position 3."),
+                Some("Invalid JSON path expression. The error is around character position 4."),
```

Comment:
The error position is a bit confusing...?

Reply:
Yep! Fixed.